src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -538,7 +541,9 @@ def update_batch_job(self, batch_id: str) -> List[str]:
                 if child_id in self._running_jobs:
                     child_job = self._running_jobs[child_id]["job"]
                 else:
-                    child_job = Job.from_attributes(job_id=child_id)
+                    child_job = Job.from_job_id(
```

Comment:
it would be more efficient to collect a list of the missing child jobs, fetch the data from ee2, and then create jobs from the results.